### PR TITLE
comdb2_tags and comdb2_tag_columns system tables.

### PR DESCRIPTION
**Pull request by dorinhogea**
per title.

### CODE REVIEW

**Review by roborivers — COMMENTED** [review]
Lint check:

```diff


@@ sqlite/ext/comdb2/tables.c @@
+ If a new system table is added, be sure to document it in docs/pages/programming/system_tables.md.
```

**Review by roborivers — CHANGES_REQUESTED** [review]
<p><a href='https://bloomberg.github.io/comdb2/contrib.html#code-style'>Coding style</a> check: <b>Error</b>. &#x26A0;.<br>Smoke testing: Success &#10003;.<br>Cbuild submission: Success &#10003;.<br>Regression testing: <b>127/521 tests failed</b> &#x26A0;.</p><p>The first 10 failing tests are: <br>queuedb_rollover_noroll1_generated<br>queuedb_rollover<br>auth_twofiles_generated<br>auth_queueodh_generated<br>auth<br>userschema_queueodh_generated<br>userschema<br>comdb2sys_queueodh_generated<br>comdb2sys_pagesize_generated<br>comdb2sys</p>

**Review by roborivers — COMMENTED** [review]
Lint check:

```diff


@@ sqlite/ext/comdb2/tables.c @@
+ If a new system table is added, be sure to document it in docs/pages/programming/system_tables.md.
```

**Review by roborivers — CHANGES_REQUESTED** [review]
<p><a href='https://bloomberg.github.io/comdb2/contrib.html#code-style'>Coding style</a> check: <b>Error</b>. &#x26A0;.<br>Smoke testing: Success &#10003;.<br>Cbuild submission: Error &#x26A0;.<br>Regression testing: <b>6/521 tests failed</b> &#x26A0;.</p><p>The first 10 failing tests are: <br>phys_rep_tiered<br>cdb2sql_queueodh_generated<br>cdb2sql<br>maxtable<br>truncatesc<br>truncatesc_ufid_on_generated</p>

**Review by roborivers — CHANGES_REQUESTED** [review]
<p><a href='https://bloomberg.github.io/comdb2/contrib.html#code-style'>Coding style</a> check: <b>Error</b>. &#x26A0;.<br>Smoke testing: Success &#10003;.<br>Cbuild submission: Success &#10003;.<br>Regression testing: <b>5/521 tests failed</b> &#x26A0;.</p><p>The first 10 failing tests are: <br>logfill<br>queuedb_rollover<br>lostwrite<br>cdb2sql_queueodh_generated<br>cdb2sql</p>

**Comment by dorinhogea** [comment]
cdb2test   Mar 11 16:50:29 2024  in-comdb2db systable_tags.R20240311.8

**Review by roborivers — COMMENTED** [review]
Lint check:

```diff


@@ sqlite/ext/comdb2/tables.c @@
+ If a new system table is added, be sure to document it in docs/pages/programming/system_tables.md.
```

**Review by roborivers — CHANGES_REQUESTED** [review]
<p><a href='https://bloomberg.github.io/comdb2/contrib.html#code-style'>Coding style</a> check: <b>Error</b>. &#x26A0;.<br>Smoke testing: Success &#10003;.<br>Cbuild submission: Success &#10003;.<br>Regression testing: <b>4/521 tests failed</b> &#x26A0;.</p><p>The first 10 failing tests are: <br>queuedb_rollover_noroll1_generated<br>cdb2sql_queueodh_generated<br>cdb2sql<br>truncatesc_ufid_on_generated</p>

File: sqlite/ext/comdb2/tags.c [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+int systblTagsInit(sqlite3 *db)
+{
+    return create_system_table(
+        db, "comdb2_tags", &systblTagsModule,
+        collect_tags, free_tags, sizeof(struct systable_tag),
+        CDB2_CSTRING, "tablename", -1, 0,
+        CDB2_CSTRING, "tagname", -1, offsetof(struct systable_tag, tagname),
+        CDB2_INTEGER, "ixnum", -1, offsetof(struct systable_tag, ixnum),
+        CDB2_INTEGER, "size", -1, offsetof(struct systable_tag, recsize),
+        CDB2_INTEGER, "columns", -1, offsetof(struct systable_tag, nmembers),
```
Comment (aakash10292) [inline sqlite/ext/comdb2/tags.c:147]: perhaps numcolumns ? to keep consistent with numblobs and numindexes below ?

Comment (dorinhogea) [reply sqlite/ext/comdb2/tags.c]: done

**Review by aakash10292 — APPROVED** [review]

**Pull request merged**